src/vtquery.cpp

```diff
@@ -188,7 +182,7 @@ struct Worker : Nan::AsyncWorker {
                     // if there are items in the data.layers vector AND
                     // the current layer name is not a part of that list, continue
                     std::string layer_name = std::string(layer.name());
-                    if (data.layers.size() > 0 && std::find(data.layers.begin(), data.layers.end(), layer_name) == data.layers.end()) {
+                    if (!data.layers.empty() && std::find(data.layers.begin(), data.layers.end(), layer_name) == data.layers.end()) {
```

Comment:
Why the `&&` here? If you do not find anything, it would be empty, and if it is empty you will not find anything.

Reply:
If it's empty, that means we want to query **all** layers instead of **no** layers.